app/controllers/api/v1/mixins/index_mixin.rb

```diff
@@ -24,28 +24,41 @@ def collection(base_query, pre_authorized: false)
         def rbac_scope(relation, pre_authorized: false)
           return relation if pre_authorized
 
-          if Catalog::RBAC::Role.catalog_administrator?
-            relation
-          else
-            access_relation(relation)
-          end
-        end
+          # if Catalog::RBAC::Role.catalog_administrator?
+          #   relation
+          # else
+          #   access_relation(relation)
+          # end
 
-        def access_relation(relation)
-          access_obj = Insights::API::Common::RBAC::Access.new(relation.model.table_name, 'read').process
-          raise Catalog::NotAuthorized, "Not Authorized for #{relation.model}" unless access_obj.accessible?
-          if access_obj.owner_scoped?
+          access_scopes = pundit_user.catalog_access.scopes(relation.model.table_name, 'read')
+
+          if access_scopes.include?('admin')
+            relation
+          elsif access_scopes.include?('group')
+            ids = Catalog::RBAC::AccessControlEntries.new.ace_ids('read', relation.model)
+            relation.where(:id => ids)
+          elsif access_scopes.include?('user')
             relation.by_owner
           else
-            if relation.model.try(:supports_access_control?)
-              ids = Catalog::RBAC::AccessControlEntries.new.ace_ids('read', relation.model)
-              relation.where(:id => ids)
-            else
-              relation
-            end
+            false
```

Comment:
@eclarizio We should raise an error here and not return a false, I think the callers are expecting a ActiveRecord relation object